### PR TITLE
Custom resolution

### LINKED ISSUE #1176: 16:9 resolution setting
I believe that we should be able to make a horizontal based video instead of just a short form video just because it would help people trying to make money once they are a little bigger from shorts and already have a fanbase.

I believe that we could make this happen by adding a setting in config.toml that is something like Full Length and it would make a video of a few different posts based on your configuration in config.toml.

I also think that we should have the option to disable the watermark since it is a considerable slowdown in the performance since 2.3 and a lot of background videos just require a link/credit from the creator in every description of the video.

### PR DESCRIPTION
**Pull request by jotonedev**
# Description

Add the option to set a custom resolution for the final video

# Issue Fixes

#1176 
#943 

# Checklist:

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

# Any other information (e.g how to test the changes)

To test this changes you need to modify the resolution values in the config file.

### CODE REVIEW

**Comment by jotonedev** [comment]
I'm still working on the option the change the background resolution.

File: video_creation/screenshot_downloader.py [hunk 1]
```diff
@@ -22,27 +21,53 @@ def download_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: in
...
         browser = p.chromium.launch(headless=True)  # add headless=False for debug
-        context = browser.new_context()
 
+        # Device scale factor (or dsf for short) allows us to increase the resolution of the screenshots
+        # When the dsf is 1, the width of the screenshot is 600 pixels
+        # so we need a dsf such that the width of the screenshot is greater than the final resolution of the video
+        dsf = (W // 600) + 1
+
+        context = browser.new_context(
+            locale=lang or "en-us",
```
Comment (OpenSourceSimon) [inline video_creation/screenshot_downloader.py:46]: Is this for the translation of the buttons in the screenshots?

**Review by OpenSourceSimon — APPROVED** [review]
Awesome work!

File: video_creation/screenshot_downloader.py (hunk 1 above)
Comment (jotonedev) [reply video_creation/screenshot_downloader.py:46]: Yes, but I still need to test it. I think it should work if the lang is an ISO code.

Comment (OpenSourceSimon) [reply video_creation/screenshot_downloader.py:46]: Nice, maybe we need to add a check for it

Comment (jotonedev) [reply video_creation/screenshot_downloader.py:46]: Okay I tested it with lang=it and it works.

**Comment by CJthisis** [comment]
I definitely wanna see this in version 2.5!

**Comment by jotonedev** [comment]
I've noticed that in backgrounds.json there is a fourth value representing the position of where to cut the background. That was never implemented in final_video.py and also it won't work because the position is relative to the resolution used for the background (That means if I want to use a 4K background the position will point to something else). So it must be completely changed or removed.

**Comment by Acalann** [comment]
Is this working as of now?

**Comment by OpenSourceSimon** [comment]
> Is this working as of now?

It is working according to another contributer. I'll test and merge this as soon as possible! Feel free to test it yourself

**Review by OpenSourceSimon — APPROVED** [review]
Tested it and looks good to me!

**Pull request merged**